presidio-analyzer/presidio_analyzer/predefined_recognizers/in_vehicle_registration_recognizer.py

```diff
@@ -0,0 +1,181 @@
+        return self.__check_vehicle_registration(sanitized_value)
+
+    def __check_vehicle_registration(self, sanitized_value: str) -> bool:
+        # print('check function called')
+        is_valid_registration = None
+        # logic here
+        state_rto_districtcode_map = {
+            "AN": Utils.in_vehicle_dist_an,
+            "AP": Utils.in_vehicle_dist_ap,
+            "AR": Utils.in_vehicle_dist_ar,
+            "AS": Utils.in_vehicle_dist_as,
+            "BR": Utils.in_vehicle_dist_br,
+            "CG": Utils.in_vehicle_dist_cg,
+            "CH": Utils.in_vehicle_dist_ch,
+            "DD": Utils.in_vehicle_dist_dd,
+            "DN": Utils.in_vehicle_dist_dn,
+            "DL": Utils.in_vehicle_dist_dl,
+            "GA": Utils.in_vehicle_dist_ga,
+            "GJ": Utils.in_vehicle_dist_gj,
+            "HP": Utils.in_vehicle_dist_hp,
+            "HR": Utils.in_vehicle_dist_hr,
+            "JH": Utils.in_vehicle_dist_jh,
+            "JK": Utils.in_vehicle_dist_jk,
+            "KA": Utils.in_vehicle_dist_ka,
+            "KL": Utils.in_vehicle_dist_kl,
+            "LA": Utils.in_vehicle_dist_la,
+            "LD": Utils.in_vehicle_dist_ld,
+            "MH": Utils.in_vehicle_dist_mh,
+            "ML": Utils.in_vehicle_dist_ml,
+            "MN": Utils.in_vehicle_dist_mn,
+            "MP": Utils.in_vehicle_dist_mp,
+            "MZ": Utils.in_vehicle_dist_mz,
+            "NL": Utils.in_vehicle_dist_nl,
+            "OD": Utils.in_vehicle_dist_od,
+            "OR": Utils.in_vehicle_dist_or,
+            "PB": Utils.in_vehicle_dist_pb,
+            "PY": Utils.in_vehicle_dist_py,
+            "RJ": Utils.in_vehicle_dist_rj,
+            "SK": Utils.in_vehicle_dist_sk,
+            "TN": Utils.in_vehicle_dist_tn,
+            "TR": Utils.in_vehicle_dist_tr,
+            "TS": Utils.in_vehicle_dist_ts,
+            "UK": Utils.in_vehicle_dist_uk,
+            "UP": Utils.in_vehicle_dist_up,
+            "WB": Utils.in_vehicle_dist_wb,
+        }
+        two_factor_registration_prefix = []
+        two_factor_registration_prefix.extend(Utils.in_union_territories)
+        two_factor_registration_prefix.extend(Utils.in_states)
+        two_factor_registration_prefix.extend(Utils.in_old_states)
+        two_factor_registration_prefix.extend(Utils.in_old_union_territories)
+        two_factor_registration_prefix.extend(Utils.in_non_standard_state_or_ut)
+        first_two_char = sanitized_value[:2].upper()
+        dist_code: str = ""
+
+        if first_two_char in two_factor_registration_prefix:
+            if sanitized_value[2].isdigit():
+                if sanitized_value[3].isdigit():
+                    dist_code = sanitized_value[2:4]
+                else:
+                    dist_code = sanitized_value[2:3]
+
+                registration_digits = sanitized_value[-4:]
+                if registration_digits.isnumeric():
+                    if 0 < int(registration_digits) <= 9999:
```

Comment:
Can we check that this is an int before casting? There could be a false positive which would fail the entire process

Reply:
line 163 is checking for that i believe. please have a look once and let me know.

Reply:
Right, sorry I missed that